### PR TITLE
[Netmanager] Run device test

### PR DESCRIPTION
**Pull request by Codebmk**
#### Summary of Changes (What does this PR do?)

- Shows an error message when the creation date is faulty ie from the future 

#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [AN-410](https://airqoteam.atlassian.net/browse/AN-410)

#### Screenshots (optional)
image

[AN-410]: https://airqoteam.atlassian.net/browse/AN-410?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Baalmart — CHANGES_REQUESTED** [review]
Hi @Codebmk , thanks for the timely response to this.


The device is the one sending a future date. See attached screenshot.
So the real "final fix" has to be done from the device firmware. cc: @deogratius2phantom @semajoel @settala1

Also, please correct the logic on the platform......if a device last sent data in the FUTURE, then why do we show XXX years AGO? Should it be showing XXX years in the FUTURE? Or something close to a clear representation of something in the future. I will let you use your creativity on this one.

**Comment by Codebmk** [comment]
Thanks Martin. I have updated the message
image

File: netmanager/src/utils/dateTime.js [hunk 1]
```diff
@@ -54,28 +53,34 @@ export const getFirstDuration = (dateTimeStr) => {
...
 
   return [year, month, day].join('-');
 };
 
+export const isDateInPast = (dateTimeStr) => {
+  let now = moment.utc();
+  let datetime = moment.utc(dateTimeStr);
+  return datetime.diff(now) < 0;
+};
+
```
Comment (Codebmk) [inline netmanager/src/utils/dateTime.js:83]: checks if the date is in the past

File: netmanager/src/views/components/DataDisplay/DeviceView/DeviceDeployStatus.js [hunk 2]
```diff
@@ -197,11 +198,25 @@ const DeviceRecentFeedView = ({ recentFeed, runReport }) => {
...
+              ) : (
+                <span className={classes.error}>in the future.</span>
+              )}
             </span>
+            {!isDateInPast(recentFeed.created_at) && (
+              <div className={classes.error}>
+                Error: Start date for this device is set to{' '}
+                {formatDateString(recentFeed.created_at)}
+              </div>
+            )}
```
Comment (Codebmk) [inline netmanager/src/views/components/DataDisplay/DeviceView/DeviceDeployStatus.js:219]: Informs user that date is in the future

Comment (Baalmart) [reply netmanager/src/views/components/DataDisplay/DeviceView/DeviceDeployStatus.js:219]: Thanks @Codebmk , please share screenshot of how the "future" message would look like?

Comment (Codebmk) [reply netmanager/src/views/components/DataDisplay/DeviceView/DeviceDeployStatus.js:219]: @Baalmart I shared it at the top but resharing here
Screenshot from 2023-05-12 07-12-57

**Review by Baalmart — APPROVED** [review]

**Pull request merged**